File: problem06.rb
def sum_of_squares num
  memo = 1
  (1..num).reduce do |memo, current|
    memo += current ** 2
  end
end

def squares_sum num
  (1..num).reduce(:+) ** 2
end

def difference num
  squares_sum(num) - sum_of_squares(num)
end

puts sum_of_squares 10
puts squares_sum 10
puts difference 10

File: problem06.js
var sum_of_squares = function(num) {
  var sum = 0;
  for (var i = 0; i <= num; i++) {
    sum += Math.pow(i, 2);
  }
  return sum;
}

var squares_sums = function(num) {
  var temp = 0;
    for (var i = 1; i <= num; i++) {
      temp +=i;
    }
  return Math.pow(temp, 2);
}

var difference = function (num) {
  return squares_sums(num) - sum_of_squares(num);
}

console.log(sum_of_squares(100));
console.log(squares_sums(100));
console.log(difference(100));
